File: module_diagnosis/src/main/java/com/zdyb/module_diagnosis/receiver/USBReceiver.java
package com.zdyb.module_diagnosis.receiver;

import android.app.Activity;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.storage.StorageManager;
import android.text.TextUtils;
import android.util.Log;

import com.zdyb.lib_common.base.BaseApplication;
import com.zdyb.lib_common.bus.BusEvent;
import com.zdyb.lib_common.bus.EventTypeDiagnosis;
import com.zdyb.lib_common.bus.RxBus;
import com.zdyb.module_diagnosis.MyApplication;

public class USBReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        StorageManager mStorageManager = (StorageManager) context.getSystemService(Activity.STORAGE_SERVICE);
        String action = intent.getAction();
        if (action.equals(Intent.ACTION_MEDIA_MOUNTED)) {
            // U盘根目录
            String mountPath = intent.getData().getPath();
            if (!TextUtils.isEmpty(mountPath)) {
                Log.d("TAG", "U盘挂载:" + mountPath);
                RxBus.getDefault().post(new BusEvent(EventTypeDiagnosis.USB_PATH,mountPath));
                BaseApplication.USBPath = mountPath;
            }
        } else if (action.equals(Intent.ACTION_MEDIA_UNMOUNTED) || action.equals(Intent.ACTION_MEDIA_EJECT)) {
            Log.d("TAG", "U盘移除");
            RxBus.getDefault().post(new BusEvent(EventTypeDiagnosis.USB_OUT));
            BaseApplication.USBPath = null;
        } else if (action.equals("android.intent.action.BOOT_COMPLETED")) {

        }


    }
}

File: lib_common/src/main/java/com/zdyb/lib_common/receiver/USBReceiver.kt
package com.zdyb.lib_common.receiver;

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.hardware.usb.UsbDevice
import android.hardware.usb.UsbManager

class USBReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context?, intent: Intent?) {

        //val usbDevice: UsbDevice? = intent!!.getParcelableExtra(UsbManager.EXTRA_DEVICE)
        if (intent != null) {
            when(intent.action){
                UsbManager.ACTION_USB_DEVICE_ATTACHED ->{
                    // 插入USB设备
                    callBack?.usbState(true)
                }
                UsbManager.ACTION_USB_DEVICE_DETACHED ->{
                    // 拔出USB设备
                    callBack?.usbState(false)
                }
            }
        }
    }

    private var callBack :CallBack? = null
    fun setListener(callBack :CallBack){
        this.callBack = callBack
    }

    public interface CallBack{
        fun usbState(boolean: Boolean)
    }

}